xarray/core/iterators.py

```diff
@@ -60,15 +63,14 @@ def __next__(self) -> Iterator[Tree]:
 
     @staticmethod
     @abstractmethod
-    def _iter(children: List[Tree], filter_, stop, maxlevel) -> Iterator[Tree]:
-        ...
+    def _iter(children: list[Tree], filter_, stop, maxlevel) -> Iterator[Tree]: ...
 
     @staticmethod
     def _abort_at_level(level, maxlevel):
         return maxlevel is not None and level > maxlevel
 
     @staticmethod
-    def _get_children(children: List[Tree], stop) -> List[Tree]:
+    def _get_children(children: list[Tree], stop) -> list[Tree]:
```

Comment:
```suggestion
    def _get_children(children: list[Tree], stop: Callable) -> list[Tree]:
```
?

Reply:
Also done in: [dc8ad9](https://github.com/pydata/xarray/pull/8879/commits/dc8ad955173effde954f47c3b380fc5bc4a7432c)